### ISSUE TITLE
[BUG] API Standardization

### ISSUE THREAD

**Issue by FBruzzesi** [koaning/scikit-lego#590]
I find that the API lacks a bit of standardization in certain contexts. By that I refer to the fact that the same attribute has different names in different classes:

## Examples

- [X] **Number of input features**:
  - In scikit-learn they are referred to as `n_features_in_`
  - In sklego sometimes it's `dim_` (e.g. [`DictMapper`](https://github.com/koaning/scikit-lego/blob/main/sklego/preprocessing/dictmapper.py#L40)), sometimes it's `n_columns_` (e.g. [`ColumnCapper`](https://github.com/koaning/scikit-lego/blob/main/sklego/preprocessing/columncapper.py#L127)), sometimes the entire `shape_` is stored (e.g. [`IdentityTransformer`](https://github.com/koaning/scikit-lego/blob/main/sklego/preprocessing/identitytransformer.py#L28)).

- [ ] **Model/Estimator param**: some classes (especially in the meta module), accept an estimator as input, sometimes this is referred to as `model` (e.g. in [`DecayEstimator`](https://github.com/koaning/scikit-lego/blob/main/sklego/meta/decay_estimator.py#L36)), sometimes as `estimator` (e.g. in [`SubjectiveClassifier`](https://github.com/koaning/scikit-lego/blob/main/sklego/meta/subjective_classifier.py#L43)).

- [X] **`__init__`**:  There are cases in which class attributes are assigned during initialization (e.g. [`RandomRegressor`](https://github.com/koaning/scikit-lego/blob/main/sklego/dummy.py#L23)).

## Proposed solution

- For attributes standardization, it should be easy enough to follow the scikit-learn API and add properties that refer to the current attributes but also add one warning (either `DeprecationWarning`, `PendingDeprecationWarning` or `FutureWarning`).
- For model/estimator keyword, something similar may be possible but I would need to double check that.
- Regarding `__init__`, it is enough to move those as (semi)private class attributes).

## Disclaimer

Not sure if the _bug_ label is the best one 🙃

**Comment by koaning** [comment]
IIRC sklearn became more strict about `n_features_in_` in recent years. But yeah, standardisation seems fine.
